keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/ParetoVertexTest.java

```diff
@@ -28,27 +34,96 @@
 
     private KeanuRandom random;
 
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
     @Before
     public void setup() {
         random = new KeanuRandom(1);
     }
 
     @Test
-    public void matchesKnownLogDensityOfScalar() {
+    public void logProbMatchesKnownLogDensityOfScalar() {
         ParetoDistribution baseline = new ParetoDistribution(1.0, 1.5);
         ParetoVertex vertex = new ParetoVertex(1.0, 1.5);
         double expected = baseline.logDensity(1.25);
         ProbabilisticDoubleTensorContract.matchesKnownLogDensityOfScalar(vertex, 1.25, expected);
     }
 
     @Test
-    public void matchesKnownLogDensityOfVector() {
+    public void logProbGraphMatchesKnownLogDensityOfScalar() {
+        DoubleVertex location = ConstantVertex.of(1.);
+        DoubleVertex scale = ConstantVertex.of(1.5);
+        ParetoVertex paretoVertex = new ParetoVertex(location, scale);
+        LogProbGraph logProbGraph = paretoVertex.logProbGraph();
+
+        LogProbGraphValueFeeder.feedValue(logProbGraph, location, location.getValue());
+        LogProbGraphValueFeeder.feedValue(logProbGraph, scale, scale.getValue());
+        LogProbGraphValueFeeder.feedValue(logProbGraph, paretoVertex, DoubleTensor.scalar(1.25));
+
+        ParetoDistribution baseline = new ParetoDistribution(1., 1.5);
+        double expected = baseline.logDensity(1.25);
+        LogProbGraphContract.matchesKnownLogDensity(logProbGraph, expected);
+    }
+
+    @Test
+    public void logProbMatchesKnownLogDensityOfVector() {
         ParetoDistribution baseline = new ParetoDistribution(1.0, 1.5);
         ParetoVertex vertex = new ParetoVertex(1.0, 1.5);
         double expected = baseline.logDensity(1.25) + baseline.logDensity(6.5);
         ProbabilisticDoubleTensorContract.matchesKnownLogDensityOfVector(vertex, new double[]{1.25, 6.5}, expected);
     }
 
+    @Test
+    public void logProbGraphMatchesKnownLogDensityOfVector() {
+        DoubleVertex location = ConstantVertex.of(1., 1.);
+        DoubleVertex scale = ConstantVertex.of(1.5, 1.5);
+        ParetoVertex paretoVertex = new ParetoVertex(location, scale);
+        LogProbGraph logProbGraph = paretoVertex.logProbGraph();
+
+        LogProbGraphValueFeeder.feedValue(logProbGraph, location, location.getValue());
+        LogProbGraphValueFeeder.feedValue(logProbGraph, scale, scale.getValue());
+        LogProbGraphValueFeeder.feedValue(logProbGraph, paretoVertex, DoubleTensor.create(1.25, 6.5));
+
+        ParetoDistribution baseline = new ParetoDistribution(1., 1.5);
+        double expected = baseline.logDensity(1.25) + baseline.logDensity(6.5);
+        LogProbGraphContract.matchesKnownLogDensity(logProbGraph, expected);
+    }
+
+    @Test
+    public void logProbGraphIsNegInfIfLocationOrScaleIsNotPositive() {
+        DoubleVertex location = ConstantVertex.of(-1., 1.);
```

Comment:
Can we do a check where location is negative and scale is positive to make sure it picks up the `and`